### PR TITLE
alembic: Create inspire branch for alembic

### PR DESCRIPTION
**Pull request by zzacharo**
## Checklist:

- [ ] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [ ] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [ ] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -225,6 +225,9 @@
         'invenio_db.models': [
             'inspire_workflows_audit = inspirehep.modules.workflows.models',
         ],
+        'invenio_db.alembic': [
+            'inspirehep = inspirehep:alembic',
+        ],
```
Comment (jacquerie) [inline setup.py]: Let's keep our entry points alphabetically sorted! This should go before `invenio_db.models`.

File: inspirehep/alembic/0da6eba8fbee_create_inspirehep_branch.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+depends_on = 'dbdbc1b19cf2'
+
+
+def upgrade():
+    """Upgrade database."""
+    op.create_table(
+        'workflows_record_sources',
+        sa.PrimaryKeyConstraint('record_id', 'source'),
+        sa.Column('source', sa.Text, default="", nullable=False),
+        sa.Column('record_id', UUIDType, sa.ForeignKey("records_metadata.id", ondelete='CASCADE'), nullable=False),
```
Comment (jacquerie) [inline inspirehep/alembic/0da6eba8fbee_create_inspirehep_branch.py]: Nit: let's use either single quotes or double quotes consistently.

**Comment by jmartinm** [comment]
We could do `Invenio` style where there is a first alembic recipe creating the branch (https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/alembic/1095cdf9f350_create_records_branch.py), then a recipe creating the initial tables

**Comment by jmartinm** [comment]
I think it's better if we create a first PR with the items mentioned above and merge it.

Then the merger PR if it has a new table it can include the recipe to create it.

**Comment by david-caro** [comment]
@jmartinm yep, we were actually kind of lost here without you :)

**Comment by ammirate** [comment]
@zzacharo so, can we push this on top of #2904 ?

**Comment by zzacharo** [comment]
@ammirate the previous recipe that I created for the merger should be moved to the merger branch or manual merger based on which branch needs the table it was creating. So, the concept is every modification will be a new revision under this one.

**Comment by zzacharo** [comment]
@jmartinm have you checked that in order to merge it?

**Comment by zzacharo** [comment]
depends on https://github.com/inveniosoftware-contrib/invenio-workflows-ui/pull/30 and https://github.com/inveniosoftware-contrib/invenio-workflows-files/pull/8 and https://github.com/inspirehep/inspire-crawler/pull/23 . For the last one should first integrate alembic.

File: tests/alembic/conftest.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+from sqlalchemy_utils.functions import create_database, database_exists
+
+
+@pytest.fixture(scope='module')
+def app():
+    """Flask application.
+
+    Creates a Flask application with a simple testing configuration,
+    then creates an application context and inside of it recreates
+    all databases and indices from the fixtures. Finally it yields,
```
Comment (jmartinm) [inline tests/alembic/conftest.py]: This `app` does not seem to recreate databases and indices?

Comment (zzacharo) [reply tests/alembic/conftest.py]: yep leftover. I am changing it.

**Comment by zzacharo** [comment]
@david-caro you can check it too and merge it if is ok.

**Comment by jacquerie** [comment]
As I wrote on Gitter:
> creating a new test suite for alembic is not a good idea
> because Travis limits us to 5 concurrent jobs
> so if you add another test suite for it you add >5 minutes to the build time
> besides, what's the need? it probably fits in integration

**Comment by zzacharo** [comment]
@jacquerie in my tests for alembic I destroy the database to check if migrations are applied as expected. Inside integration suite they all share the same database and things are getting messy because the app fixture is persistent across the test session. Any thoughts?

**Comment by jmartinm** [comment]
👍 I would approve it again. Nice that you could make it work inside the integration tests.

File: tests/integration/conftest.py [hunk 4]
```diff
@@ -121,6 +121,22 @@ def small_app():
         yield app
 
 
+@pytest.fixture(scope='module')
+def alembic_app():
+    """Flask application with no records and module scope."""
+    app = create_app(
+        SQLALCHEMY_DATABASE_URI='postgresql+psycopg2://inspirehep:dbpass123@localhost:5432/inspirehep_alembic'
```
Comment (david-caro) [inline tests/integration/conftest.py]: 👍 awesome xd

**Pull request merged**